torch/nn/functional.py

```diff
@@ -3510,6 +3510,170 @@ def cross_entropy(
+        return handle_torch_function(
+            linear_cross_entropy,
+            (input, linear_weight, target, linear_bias),
+            input,
+            linear_weight,
+            target,
+            linear_bias=linear_bias,
+            reduction=reduction,
+            ignore_index=ignore_index,
+            label_smoothing=label_smoothing,
+            chunking_strategy=chunking_strategy,
+            vocab_chunk_size=vocab_chunk_size,
+            batch_chunk_size=batch_chunk_size,
+        )
+
+    if not isinstance(reduction, str):
+        if hasattr(reduction, "node"):
+            from torch.fx.proxy import TraceError
+
+            raise TraceError(
+                "symbolically traced variables cannot be used as inputs to control flow"
+            )
+        raise ValueError(
+            f"reduction must be one of ('mean', 'sum', 'none'), got '{reduction}'"
+        )
+
+    if reduction not in ("mean", "sum", "none"):
+        raise ValueError(
+            f"reduction must be one of ('mean', 'sum', 'none'), got '{reduction}'"
+        )
+
+    if not (0.0 <= label_smoothing <= 1.0):
+        raise ValueError(
+            f"label_smoothing must be between 0.0 and 1.0, got {label_smoothing}"
+        )
+
+    if chunking_strategy not in ("vocab", "batch", "none"):
+        raise ValueError(
+            "chunking_strategy must be one of ('vocab', 'batch', 'none'), "
+            f"got '{chunking_strategy}'"
+        )
+
+    if vocab_chunk_size is not None and vocab_chunk_size <= 0:
+        raise ValueError(
+            f"vocab_chunk_size must be positive when provided, got {vocab_chunk_size}"
+        )
+
+    if batch_chunk_size is not None and batch_chunk_size <= 0:
+        raise ValueError(
+            f"batch_chunk_size must be positive when provided, got {batch_chunk_size}"
+        )
+
+    if (
+        input.device.type != "cpu"
+        or linear_weight.device != input.device
+        or target.device != input.device
+        or (linear_bias is not None and linear_bias.device != input.device)
+    ):
+        result = _linear_cross_entropy_naive(
+            input,
+            linear_weight,
+            target,
+            linear_bias,
+            reduction,
+            ignore_index,
+            label_smoothing,
+        )
+    else:
+        op = torch.ops.aten.linear_cross_entropy.default
+        schema = op._schema
+        supports_chunk_sizes = any(
+            arg.name == "vocab_chunk_size" for arg in schema.arguments
+        )
+        if not supports_chunk_sizes:
+            result = _linear_cross_entropy_naive(
+                input,
+                linear_weight,
+                target,
+                linear_bias,
+                reduction,
+                ignore_index,
+                label_smoothing,
+            )
+        else:
+            reduction_enum = _Reduction.get_enum(reduction)
+            try:
+                result = torch.ops.aten.linear_cross_entropy(
```

Comment:
the way to call this following the convention in this file would be `torch._C._nn.linear_cross_entropy`